back/app/controllers/omniauth_callback_controller.rb

```diff
@@ -69,6 +70,12 @@ def auth_callback(verify: false, authver_method: nil)
     end
 
     if @user
+      if @user.invite_pending? && auth["info"]["email_verified"] == false
+        # If the user is invited but the SSO email is not verified, we cannot proceed
```

Comment:
This is truly an edge case, but sometimes the email returned by SSO is not confirmed. This could, in theory, allow someone to hijack an admin account by putting e.g. the email of the admin in their FranceConnect profile. This line of code blocks this by requiring the email to be confirmed.